test/plugin/base_test.rb

```diff
@@ -116,6 +116,26 @@ def test_configure_partial_success
     end
   end
 
+  def test_metadata_agent_url_customization
+    [
+      # If @metadata_agent_url is customized, use that even if the environment
+      # variable is set.
+      [CUSTOM_METADATA_AGENT_URL_CONFIG, true, CUSTOM_METADATA_AGENT_URL],
```

Comment:
We don't seem to be testing the "config without environment variable" case. Might as well add it, just to rule out an inverted conditional...

Reply:
Added.